modules/data/redis-cache/main.bicep

```diff
@@ -0,0 +1,250 @@
+@allowed([
+  'ConnectedClientList'
+])
+param logsToEnable array = [
+  'ConnectedClientList'
+]
+
+@description('The name of metrics that will be streamed.')
+@allowed([
+  'AllMetrics'
+])
+param metricsToEnable array = [
+  'AllMetrics'
+]
+
+@description('Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID')
+param roleAssignments array = []
+
+@description('Optional.  Firewall rule for the redis cache')
+param redisFirewallRules object = {}
+
+@description('Define Private Endpoints that should be created for Azure Redis Cache.')
+param privateEndpoints array = []
+
+@description('Toggle if Private Endpoints manual approval for Azure Redis Cache should be enabled.')
+param privateEndpointsApprovalEnabled bool = false
+
+var diagnosticsLogs = [for log in logsToEnable: {
+  category: log
+  enabled: true
+  retentionPolicy: {
+    enabled: true
+    days: diagnosticLogsRetentionInDays
+  }
+}]
+
+var diagnosticsMetrics = [for metric in metricsToEnable: {
+  category: metric
+  timeGrain: null
+  enabled: true
+  retentionPolicy: {
+    enabled: true
+    days: diagnosticLogsRetentionInDays
+  }
+}]
+
+var varRedisFirewallRules = [for redisFirewallRule in items(redisFirewallRules): {
+  redisCacheFirewallRuleName: redisFirewallRule.key
+  startIP: redisFirewallRule.value.startIP
+  endIP: redisFirewallRule.value.endIP
+}]
+
+var varPrivateEndpoints = [for privateEndpoint in privateEndpoints: {
+  name: '${privateEndpoint.name}-${redisCache.name}'
+  privateLinkServiceId: redisCache.id
+  groupIds: [
+    'redisCache'
+  ]
+  subnetId: privateEndpoint.subnetId
+  privateDnsZones: contains(privateEndpoint, 'privateDnsZoneId') ? [
+    {
+      name: 'default'
+      zoneId: privateEndpoint.privateDnsZoneId
+    }
+  ] : []
+}]
+
+resource redisCache 'Microsoft.Cache/redis@2022-06-01' = {
+  name: name
+  location: location
+  tags: tags
+  properties: {
+    enableNonSslPort: enableNonSslPort
+    minimumTlsVersion: minimumTlsVersion
+    publicNetworkAccess: !empty(publicNetworkAccess) ? any(publicNetworkAccess) : (!empty(publicNetworkAccess) ? 'Disabled' : null)
+    redisConfiguration: !empty(redisConfiguration) ? redisConfiguration : null
+    redisVersion: redisVersion
+    shardCount: skuName == 'Premium' ? shardCount : null 
+    replicasPerMaster: skuName == 'Premium' ? replicasPerMaster : null
+    replicasPerPrimary: skuName == 'Premium' ? replicasPerPrimary : null
+    sku: {
+      capacity: capacity
+      family: skuName == 'Premium' ? 'P' : 'C'
+      name: skuName
+    }
+    staticIP: !empty(staticIP) ? staticIP : null
+    subnetId: !empty(subnetId) ? subnetId : null
+    tenantSettings: tenantSettings
+  }
+  zones: skuName == 'Premium' ? pickZones('Microsoft.Cache', 'redis', location, 1) : null
+}
```

Comment:
```suggestion
var isPremium = skuName == 'Premium'
resource redisCache 'Microsoft.Cache/redis@2022-06-01' = {
  name: name
  location: location
  tags: tags
  properties: {
    enableNonSslPort: enableNonSslPort
    minimumTlsVersion: minimumTlsVersion
    publicNetworkAccess: !empty(publicNetworkAccess) ? any(publicNetworkAccess) : (!empty(publicNetworkAccess) ? 'Disabled' : null)
    redisConfiguration: !empty(redisConfiguration) ? redisConfiguration : null
    redisVersion: redisVersion
    shardCount: isPremium ? shardCount : null 
    replicasPerMaster: isPremium ? replicasPerMaster : null
    replicasPerPrimary: isPremium ? replicasPerPrimary : null
    sku: {
      capacity: capacity
      family: isPremium ? 'P' : 'C'
      name: skuName
    }
    staticIP: !empty(staticIP) ? staticIP : null
    subnetId: !empty(subnetId) ? subnetId : null
    tenantSettings: tenantSettings
  }
  zones: isPremium ? pickZones('Microsoft.Cache', 'redis', location, 1) : null
}
```

Reply:
good place to use a var to make it a little easier to read the resource.

Reply:
Done,  Great Idea.